.github/workflows/azure_e2e_release_promotion_tests.yml

```diff
@@ -0,0 +1,190 @@
+name: Azure E2E Release Promotion Tests
+'on':
+  push:
+    branches:
+      - aj-1436-e2e-staging
+  workflow_dispatch:
+    inputs:
+      testEnv:
+        description: >-
+          Environment in which tests should be run. Currently only runs on
+          staging
+env:
+  BEE_NAME: >-
+    ${{ github.event.repository.name }}-${{ github.run_id }}-${{
+    github.run_attempt }}-dev
+  TOKEN: '${{ secrets.BROADBOT_TOKEN }}'
+  BEE_CREATE_RUN_NAME: >-
+    bee-create-${{ github.event.repository.name }}-${{ github.run_id }}-${{
+    github.run_attempt }}
+  ATTACH_BP_TO_LZ_RUN_NAME: >-
+    attach-billing-project-to-landing-zone-${{ github.event.repository.name
+    }}-${{ github.run_id }}-${{ github.run_attempt }}
+jobs:
+  init-github-context:
+    runs-on: ubuntu-latest
+    outputs:
+      branch: '${{ steps.extract-inputs.outputs.branch }}'
+      delete-bee: '${{ steps.extract-inputs.outputs.delete-bee }}'
+    steps:
+      - name: Get inputs or use defaults
+        id: extract-inputs
+        run: >
+          echo "branch=${{ inputs.branch || 'develop' }}" >> "$GITHUB_OUTPUT"
+
+          echo "delete-bee=${{ inputs.delete-bee || false }}" >>
+          "$GITHUB_OUTPUT"
+  params-gen:
+    runs-on: ubuntu-latest
+    outputs:
+      project-name: '${{ steps.gen.outputs.project_name }}'
+      bee-name: '${{ env.BEE_NAME }}'
+    steps:
+      - uses: actions/checkout@v4
+      - name: Generate a random billing project name
+        id: gen
+        run: |
+          project_name=$(echo "tmp-billing-project-$(uuidgen)" | cut -c -30)
+          echo "project_name=${project_name}" >> $GITHUB_OUTPUT
+  create-bee-workflow:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Echo Leo version
+        run: >
+          echo '${{ needs.leo-build-tag-publish-job.outputs.custom-version-json
+          }}'
+      - name: dispatch to terra-github-workflows
```

Comment:
in WDS GAH for e2e tests, we have [retries ](https://github.com/DataBiosphere/terra-workspace-data-service/blob/main/.github/workflows/run-e2e-tests.yaml#L52)for BEE creation and attachment of billing project, I think it may be worth it to do this here as well. 

this gets me wondering if it would be better in the future to define the bee creation/lz attachment as actions in the repo where e2e tests live and then have WDS and Leo check those out. That way in theory we dont need to edit lots of different places if those actions ever change. (this would be a follow up ticket, dont think its urgent to do now)

Reply:
but then again, both create bee and attach landing zone have another action that controls what they actually do....so not sure. maybe retries should happen in those actions that dev ops owns vs us implementing them?

Reply:
Yeah I think in theory there are plans to make create bee and attach landing zone more robust in general, making the retries a bit less necessary.